### PR TITLE
ci: add trivy workflow

### PR DESCRIPTION
**Pull request by mightyBroccoli**
to ensure critical and high vulnerabilties are detected quickly

### CODE REVIEW

**Comment by mightyBroccoli** [comment]
@dani-garcia would you mind reviewing this pr?

File: .github/workflows/trivy.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    permissions:
+      contents: read
+      security-events: write
+      actions: read
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@fbd16365eb88e12433951383f5e99bd901fc618f # v0.12.0
```
Comment (dani-garcia) [inline .github/workflows/trivy.yml:32]: This can be updated to 0.13.1: https://github.com/aquasecurity/trivy-action/releases/tag/0.13.1

**Review by dani-garcia — COMMENTED** [review]
We haven't used code scanning much in the past because in my experience we end up dealing with more false positives and non-actionable reports than good reports, but I think it's worth trying at least. Just gotta update the trivy action as there was a new release last week

File: .github/workflows/trivy.yml (hunk 1 above)
Comment (mightyBroccoli) [reply .github/workflows/trivy.yml]: done :+1:

**Review by dani-garcia — APPROVED** [review]

**Pull request merged**